### ISSUE TITLE
[2.9.0-M3] incorrect MySQL/MariaDB jdbc url handling

### ISSUE THREAD

**Issue by xuwei-k** [playframework/playframework#11744]
### Play Version

2.9.0-M3

### API

Scala

### Operating System

all

### JDK

all

### Reproducible Test Case

```scala
Welcome to Scala 2.13.10 (OpenJDK 64-Bit Server VM, Java 11.0.18).
Type in expressions for evaluation. Or try :help.

scala> val RegexOld = "^(jdbc:)?mysql:.*".r
val RegexOld: scala.util.matching.Regex = ^(jdbc:)?mysql:.*

scala> val RegexNew = "^(jdbc:)?(mysql|mariadb):.*".r
val RegexNew: scala.util.matching.Regex = ^(jdbc:)?(mysql|mariadb):.*

scala> val url = "jdbc:mysql://127.0.0.1:3306/foo"
val url: String = jdbc:mysql://127.0.0.1:3306/foo

scala> val RegexOld(_) = url // success

scala> val RegexNew(_) = url // fail
scala.MatchError: jdbc:mysql://127.0.0.1:3306/foo (of class java.lang.String)
  ... 32 elided

scala> val RegexNew(_, _) = url // success
```

### Note 

- https://github.com/playframework/playframework/pull/11655

fix regex or following change 🤔 

```diff
--- a/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/ApplicationEvolutions.scala
+++ b/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/ApplicationEvolutions.scala
@@ -280,7 +280,7 @@ private object ApplicationEvolutions {
     val (selectScript, createScript, insertScript) = url match {
       case OracleJdbcUrl() =>
         (SelectPlayEvolutionsLockOracleSql, CreatePlayEvolutionsLockOracleSql, InsertIntoPlayEvolutionsLockOracleSql)
-      case MysqlJdbcUrl(_) =>
+      case MysqlJdbcUrl(_, _) =>
         (SelectPlayEvolutionsLockMysqlSql, CreatePlayEvolutionsLockMysqlSql, InsertIntoPlayEvolutionsLockMysqlSql)
       case _ =>
         (SelectPlayEvolutionsLockSql, CreatePlayEvolutionsLockSql, InsertIntoPlayEvolutionsLockSql)
@@ -304,7 +304,7 @@ private object ApplicationEvolutions {
       attempts: Int = 5
   ): Unit = {
     val lockScripts = url match {
-      case MysqlJdbcUrl(_) => lockPlayEvolutionsLockMysqlSqls
+      case MysqlJdbcUrl(_, _) => lockPlayEvolutionsLockMysqlSqls
       case OracleJdbcUrl() => lockPlayEvolutionsLockOracleSqls
       case _               => lockPlayEvolutionsLockSqls
     }
diff --git a/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/EvolutionsApi.scala b/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/EvolutionsApi.scala
index c574edcef..85044bcec 100644
--- a/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/EvolutionsApi.scala
+++ b/persistence/play-jdbc-evolutions/src/main/scala/play/api/db/evolutions/EvolutionsApi.scala
@@ -452,7 +452,7 @@ class DatabaseEvolutions(
         val createScript = database.url match {
           case SqlServerJdbcUrl() => CreatePlayEvolutionsSqlServerSql
           case OracleJdbcUrl()    => CreatePlayEvolutionsOracleSql
-          case MysqlJdbcUrl(_)    => CreatePlayEvolutionsMySql
+          case MysqlJdbcUrl(_, _) => CreatePlayEvolutionsMySql
           case DerbyJdbcUrl()     => CreatePlayEvolutionsDerby
           case HsqlJdbcUrl()      => CreatePlayEvolutionsHsql
```

**Comment by mkurz** [comment]
Thanks! Your fixed probably makes most sense, see #11745

**Comment by mkurz** [comment]
@xuwei-k I published `2.9.0-M4` containing the fix.

**Comment by xuwei-k** [comment]
thanks! 👍
